File: weather-app/src/main/java/rcgonzalezf/org/weather/common/analytics/AnalyticsObserver.java
package rcgonzalezf.org.weather.common.analytics;

public interface AnalyticsObserver {

  void onScreen(String screenName, AnalyticsBaseData analyticsBaseData);

  void onAction(AnalyticsEvent analyticsEvent, String screenName, AnalyticsBaseData analyticsBaseData);
}

File: weather-app/src/main/java/rcgonzalezf/org/weather/common/analytics/AnalyticsObserver.kt
package rcgonzalezf.org.weather.common.analytics

interface AnalyticsObserver {
    fun onScreen(screenName: String?, analyticsBaseData: AnalyticsBaseData?)
    fun onAction(analyticsEvent: AnalyticsEvent?, screenName: String?, analyticsBaseData: AnalyticsBaseData?)
}
